contrib/native/client/example/pooledConnections.cpp

```diff
@@ -0,0 +1,300 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <fstream>
+#include <iostream>
+#include <stdio.h>
+#include <stdlib.h>
+#include <boost/thread.hpp>
+#include "drill/drillc.hpp"
+
+int nOptions=5;
+
+struct Option{
+    char name[32];
+    char desc[128];
+    bool required;
+}qsOptions[]= {
+    {"query", "Query strings, separated by semicolons", true},
+    {"connectStr", "Connect string", true},
+    {"logLevel", "Logging level [trace|debug|info|warn|error|fatal]", false},
+    {"numConnections", "Number of simultaneous connections", true},
+    {"numIterations", "Number of iterations to run. Each query is sent to each connection this many times", true}
+};
+
+std::map<std::string, std::string> qsOptionValues;
+
+const char* exceptionInject="alter session set `drill.exec.testing.controls` = '{ \"injections\" : [{ \"type\":\"exception\",\"siteClass\":\"org.apache.drill.exec.work.fragment.FragmentExecutor\",\"desc\":\"fragment-execution\",\"nSkip\":0,\"nFire\":1,\"exceptionClass\":\"java.lang.OutOfMemoryError\"}]}'";
```

Comment:
Actually, I'm going to remove this example. It is not correct as it depends on setting session options which do not work for pooled connections.